### PR TITLE
atmel/nand-controller.c: ECC statistics corrected counter update added.

### PR DESCRIPTION
**Pull request by ksdd**
During work on some issues related to ECC error correction, I discovered that the Atmel NAND controller driver never updates the corrected counter, although in my test case there have been ECC corrections.

### CODE REVIEW

**Comment by ksdd** [comment]
Hi, anyone interested in this MR?

**Comment by alexandrebelloni** [comment]
I talked with the upstream NAND maintainer, he is interested and you should probably send this patch on the mailing lists.

**Comment by ambarus** [comment]
Hi, @ksdd !

Sorry, somehow I missed this. The change looks good. Would you add a commit message and send it upstream?
You'll need {} around else too. Run "scripts/checkpatch --strict patch" before sending it upstream.

**Review by ambarus — CHANGES_REQUESTED** [review]
Looks good! Please add brackets around else, and add a commit message. Would be great if you can send it to upstream.

**Comment by ksdd** [comment]
Hi @ambarus , I added the braces to the else-branch. Hopefully it is okay now. However, I never send patches upstream myself. Up to now linux-at91 at github was my (direct) upstream. Any guidance is welcome! Do I need to create another MR? If yes, against which repo? Do I need to create a patch set? If yes, where to send it? Any pointer to a procedure describing document is welcome as well. I am a total greenhorn, when it comes to sending patches upstream.

**Comment by ambarus** [comment]
Hi, @ksdd!

The linux-mtd repo is at git://git.kernel.org/pub/scm/linux/kernel/git/mtd/linux.git. Base your patch on top of nand/next branch.
Here is the guideline on how to submit the patch upstream: https://www.kernel.org/doc/html/latest/process/submitting-patches.html

Let me know if I can further help.

**Comment by ambarus** [comment]
Hi, @ksdd!

Let me know if you want me to send on your behalf the patch to upstream.

ta

**Comment by ksdd** [comment]
Hi @ambarus,

I have trouble getting the "git send-email" workflow properly set up. Our email server seems to add some unwanted stuff to the emails.  Replies on test emails sent as described on git-send-email.io are complaining about application/pkcs7-signature attachments. I have to invoke IT department  to get this fixed. I don't know, if and when this will happen. If You like, You may send the patch to upstream.

Cheers,
Kai

**Comment by ksdd** [comment]
Hi @ambarus ,

it is difficult to get the "git send-email' properly set up in our company. Our email server automatically adds an application/pkcs7-signature attachment, which is according to my tests following the tutorial of git-send-email.io is a problem. I have to discuss with IT, how to get rid of it. So, probably it is faster You send the patch to upstream meanwhile.

Cheers,
Kai

**Comment by ambarus** [comment]
Sent patch to linux-mtd:
https://patchwork.ozlabs.org/project/linux-mtd/patch/20210322150714.101585-1-tudor.ambarus@microchip.com/

We will close this PR after the patch gets merged into linux-mtd, and after we'll get notifications that it will be integrated in the stable kernels. Once in the stable kernels we will get the patch in linux-at91 when pulling new stable updates.

**Comment by cristibirsan** [comment]
The patch was accepted upstream and is also added to linux-5.10-at91 branch so I'll close it.

**Pull request closed without merging**
